lib/moon_web/pages/usage_page.ex

```diff
@@ -0,0 +1,122 @@
+          <a
+            class="text-piccolo-100 font-medium transition-colors duration-200 hover:text-hit visited:text-hit"
+            href="https://surface-ui.org/documentation"
+            rel="nofollow"
+            target="_blank"
+          >
+            Surface documentation
+          </a>
+          for detailed instructions about components. Just alias them (if you want to) and insert
+          into the templates like this:
+        </p>
+        <CodeSnippet>{surface_code()}</CodeSnippet>
+      </PageSection>
+      <PageSection title="HEEx templates">
+        <p>
+          But anyway you still can use Moon components with LiveView mechanism only. Some helpers are made for this.
+          But please note, that only some Surface features are supported.
+          We've done even a bit more, than mentioned in
+          <a
+            class="text-piccolo-100 font-medium transition-colors duration-200 hover:text-hit visited:text-hit"
+            href="https://surface-ui.org/usage_with_phoenix_templates"
+            rel="nofollow"
+            target="_blank"
+          >
+            Surface's guides
+          </a>
+          that actually do not work, except recommended way - rewrite all stuff as Surface's views and components.
+          <p>
+          </p>
+          So, working code looks like:
+        </p>
+        <CodeSnippet>{liveview_code()}</CodeSnippet>
+        <p>And for the contexts - You can pass context as a Map to the component if needed, but context isn't populated to child components within HEEX template, sorry.</p>
+        <CodeSnippet>{liveview_context_code()}</CodeSnippet>
+      </PageSection>
+    </Page>
+    """
+  end
+
+  def surface_code() do
+    """
+    alias Moon.Components.Button
+
+    def render(assigns) do
+      ~F\"""
+        ...
+          <Button variant="secondary">Secondary</Button>
+        ...
+      \"""
+    end
+    """
+  end
+
+  def liveview_code() do
+    """
+    alias Moon.Components.Button
+    alias Moon.Components.Datepicker
+
+    import Moon.Helpers.SurfaceRender
+
+    def render(assigns) do
+      ~H\"""
+        ...
+          <.surface module={Button} variant="secondary">Secondary</.surface>
+
+          # or, if you wish more explicit naming - use for stateless components
+          <.surface_component module={Button} variant="secondary">Secondary</.surface>
+
+          # and for stateful component
+          <.surface_live_component module={Datepicker}
+            start_date={Timex.today() |> Timex.beginning_of_week()}
+            end_date={Timex.today() |> Timex.end_of_week()}
+            week_starts_on={7}
+            id="datepicker"
+          />
+        ...
+      \"""
+    end
+    """
+  end
+
+  def liveview_context_code() do
+    """
+    def render(assigns) do
+      ~H\"""
+        ...
+          <.surface module={Button}
+              variant="secondary"
+              __context__={%{locale: @locale}}
```

Comment:
This should be outer component?

Reply:
And can we rename .surface to .moon and .moon_component and .moon_live_component?

Reply:
outer context component (with `Surface.Components.Context`) does not work ((((
arbitary props are set as usual - `<.surface module={Button} {my_props}>hit me</.surface>` when  is_list(my_props)  or is_map(my_props)

Reply:
renaming is done

Reply:
@phgrey There is no info, how Context would work?

https://github.com/surface-ui/surface/blob/main/lib/surface/components/context.ex